core/shared/src/main/scala/scalaz/zio/internal/Env.scala

```diff
@@ -39,11 +39,8 @@ trait Env {
   /**
    * Awaits for the result of the fiber to be computed.
    */
-  final def unsafeRun[E, A](io: IO[E, A]): A = {
-    val exit = unsafeRunSync(io)
-
-    exit.fold(cause => throw new FiberFailure(cause), identity)
-  }
+  final def unsafeRun[E, A](io: IO[E, A]): A =
+    unsafeRunSync(io).fold(cause => throw new FiberFailure(cause), identity)
```

Comment:
Sexy:
```scala
unsafeRunSync(io).getOrThrow(new FiberFailure(_))
```

Reply:
To avoid having a partial function in the public API, I added `getOrElse` instead, which makes for something nearly as sexy:

```scala
unsafeRunSync(io).getOrElse(throw new FiberFailure(_))
```